crates/build/re_types_builder/src/codegen/cpp/mod.rs

```diff
@@ -1576,6 +1576,7 @@ fn to_arrow_method(
 
 fn archetype_serialize(type_ident: &Ident, obj: &Object, hpp_includes: &mut Includes) -> Method {
     hpp_includes.insert_rerun("component_batch.hpp");
+    hpp_includes.insert_rerun("component_descriptor.hpp");
```

Comment:
we actually don't use `ComponentDescriptor` in the header of the archetype at all.
We do use it in the cpp, but we already include the types for concrete components there anyways, so we know about `ComponentDescriptor` transitively

```suggestion
```